doc/user/sideeffect.xml

```diff
@@ -46,79 +46,157 @@
+ The primary use for the &SideEffect; method
  is to prevent two build steps from simultaneously modifying the same file.
 
  </para>
 
  <para>
- TODO: currently doesn't work due to issue #2154:
- https://github.com/SCons/scons/issues/2154
+
+ In this example, the rule to build <filename>file1</filename>
+ will also put data into <filename>log</filename>, which is used
+ as a source for the command to generate <filename>file2</filename>,
+ but <filename>log</filename> is unknown to &SCons; on a clean
+ build: it neither exists, nor is it the target of any builder.
+ The <filename>SConscript</filename> uses
+ &SideEffect; to inform &SCons; about the additional output file.
+
  </para>
- 
+
+ <scons_example name="sideeffect_simple">
+   <file name="SConstruct" printme="1">
+env = Environment()
+f2 = env.Command(
+    target='file2',
+    source='log',
+    action=Copy('$TARGET', '$SOURCE')
+)
+f1 = env.Command(
+    target='file1',
+    source=[],
+    action='echo >$TARGET data1; echo >log updated file1'
+)
+env.SideEffect('log', f1)
+   </file>
+ </scons_example>
+
  <para>
 
- If more than one build step creates or manipulates the same file,
- it can cause unpleasant results if both build steps are run at the same time.
- The shared file is declared as a side-effect of building the primary targets
- and &SCons; will prevent the two build steps from running in parallel.
+ Without the &f-SideEffect;, this build would fail with a message
+ <computeroutput>Source `log' not found, needed by target `file2'</computeroutput>,
+ but now it can proceed:
 
  </para>
 
+ <scons_output example="sideeffect_simple" suffix="1">
+    <scons_output_command>scons -Q</scons_output_command>
+ </scons_output>
+
  <para>
 
- In this example, the <filename>SConscript</filename> uses
- &SideEffect; to inform &SCons; about the additional output file.
+ However, it is better to actually identify
+ <filename>log</filename> as a target, since in this
+ case that's what it is:
 
  </para>
 
- <scons_example name="sideeffect_simple">
+ <scons_example name="sideeffect_simple2">
    <file name="SConstruct" printme="1">
 env = Environment()
-f2 = env.Command('file2', 'log', Copy('$TARGET', '$SOURCE'))
-f1 = env.Command('file1', [],
-    'echo >$TARGET data1; echo >log updated file1')
-env.SideEffect('log', env.Command('file1', [],
-    'echo >$TARGET data1; echo >log updated file1'))
+f2 = env.Command(
+    target='file2',
+    source='log',
+    action=Copy('$TARGET', '$SOURCE')
+)
+f1 = env.Command(
+    target=['file1', 'log'],
+    source=[],
+    action='echo >$TARGET data1; echo >log updated file1'
+)
    </file>
  </scons_example>
 
+ <scons_output example="sideeffect_simple2" suffix="1">
+    <scons_output_command>scons -Q</scons_output_command>
+ </scons_output>
+
  <para>
 
- Even when run in parallel mode, &SCons; will run the two steps in order:
+ In general, &SideEffect; is not intended for the case when
+ a command produces extra files. For example, the
```

Comment:
extra target files (files which will be used as sources to other build steps).

Reply:
I updated this and tweaked some nearby wording.